istioctl/cmd/proxyconfig.go

```diff
@@ -116,13 +120,26 @@ var stringToLevel = map[string]Level{
 }
 
 var (
-	loggerLevelString = ""
-	reset             = false
+	loggerLevelString      = ""
+	reset                  = false
+	once                   sync.Once
+	isZtunnelPodKubeClient kube.CLIClient
 )
 
 var isZtunnelPod = func(podName, podNamespace string) bool {
-	client, _ := kubeClient(kubeconfig, configContext)
-	return ambientutil.IsZtunnelPod(client, podName, podNamespace)
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+	var err error
+
+	b := backoff.NewExponentialBackOff(backoff.DefaultOption())
+	once.Do(func() {
+		err = b.RetryWithContext(ctx, func() error {
+			isZtunnelPodKubeClient, err = kubeClient(kubeconfig, configContext)
+			log.Warnf("failed to create k8s client")
+			return err
```

Comment:
Just noticed this error got swallowed. You probably are better off just adding an error to the method signature; from a quick glance at callers, it looks like most can handle returning an error